### PR TITLE
Add aspnetapp sample

### PR DESCRIPTION
**Pull request by kendrahavens**
This sample will be a useful reference for Docker documentation on .NET.

Suggested reviewers
@glennc @richlander @MichaelSimons

### CODE REVIEW

**Comment by kendrahavens** [comment]
@glenn @MichaelSimons @spboyer @richlander If this looks good can I get a LGTM?

**Comment by MichaelSimons** [comment]
I think the same comments I made about the docker sample apply here.  I think the .dockerignore can be simplified which eliminates the duplicated going into the image.

**Comment by glennc** [comment]
Do we need this to be a sample with Auth? If Auth is not important for the sample then creating a new app without Auth (which removes the need for a database) reduces lots of the noise. It also means you are not using SQLite, which would be nice since you can't realistically do that for this sort of data. Alternatively just doing social auth or something else that removes the database requirement to simplify the sample app for people learning.

The Dockerfiles and such would all be the same.

**Comment by kendrahavens** [comment]
@MichaelSimons I was able to simplify the .dockerignore file with your suggestions on changing the Dockerfile. Thank you!

@glennc It's now a no auth app. Sorry, I had a hard time figuring out what files actually made it auth. I didn't realize the default `dotnet new mvc` had auth.

File: aspnetapp/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*
```
Comment (MichaelSimons) [inline aspnetapp/.dockerignore:1]: Looks Great!

File: aspnetapp/Dockerfile.nano [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM microsoft/dotnet:nanoserver
```
Comment (MichaelSimons) [inline aspnetapp/Dockerfile.nano]: I think this should be a versioned image and it shouldn't rely on an sdk image.  I think 1.1-runtime-nanoserver would be more appropriate.

File: aspnetapp/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM microsoft/aspnetcore:1.1.0
```
Comment (MichaelSimons) [inline aspnetapp/Dockerfile]: I think referencing 1.1 would be better.  I see no reason to be tied to 1.1.0.

File: README.md [hunk 4]
```diff
@@ -31,8 +31,9 @@ The following samples show different ways to use .NET Core images.
...
-- [dotnetapp-current](dotnetapp-current) -- This sample demonstrates how to configure an application to use the .NET Core 1.1 image. Both project.json and the Dockerfile have been updated to depend on .NET Core 1.1. This sample is the same as [dotnetapp-prod](dotnetapp-prod) with the exception of relying on a later .NET Core version.
+- [dotnetapp-selfcontained](dotnetapp-selfcontained) - This sample is also good for production scenarios since it relies on an operating system image (without .NET Core). [Self-contained .NET Core apps](https://docs.microsoft.com/dotnet/articles/core/deploying/) include .NET Core as part of the app and not as a centrally installed component in a base image.
+- [dotnetapp-current](dotnetapp-current) - This sample demonstrates how to configure an application to use the .NET Core 1.1 image. Both project.json and the Dockerfile have been updated to depend on .NET Core 1.1. This sample is the same as [dotnetapp-prod](dotnetapp-prod) with the exception of relying on a later .NET Core version.
```
Comment (MichaelSimons) [inline README.md]: Looks like I forgot to update this - could you update the `project.json` reference to something like `the project file`

**Review by MichaelSimons — COMMENTED** [review]
Looks good - I have just a couple comments.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**